homeassistant/components/conversation/default_agent.py

```diff
@@ -733,10 +732,13 @@ def _make_slot_lists(self) -> dict[str, SlotList]:
             return self._slot_lists
 
         entity_registry = er.async_get(self.hass)
+        to_expose = async_should_expose_entities(
+            self.hass, DOMAIN, self.hass.states.async_entity_ids()
+        )
         states = [
             state
             for state in self.hass.states.async_all()
-            if async_should_expose(self.hass, DOMAIN, state.entity_id)
+            if state.entity_id in to_expose
         ]
```

Comment:
There is no await here so there is no risk of the state machine changing in between calls